hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -348,12 +352,22 @@ private void handleUserTransaction(
                     fees,
                     platformEvent.getCreatorId().id());
             if (validationResult.status() != SO_FAR_SO_GOOD) {
-                if (validationResult.status() == NODE_DUE_DILIGENCE_FAILURE) {
-                    payer = creator.accountId();
-                }
-                final var penaltyFee = new Fees(fees.nodeFee(), fees.networkFee(), 0L);
-                feeAccumulator.charge(payer, penaltyFee);
                 recordBuilder.status(validationResult.responseCodeEnum());
+                try {
+                    final var penaltyPayerID =
+                            validationResult.status() == NODE_DUE_DILIGENCE_FAILURE ? creator.accountId() : payer;
+                    final var penaltyFee = new Fees(fees.nodeFee(), fees.networkFee(), 0L);
+                    feeAccumulator.charge(penaltyPayerID, penaltyFee);
+                } catch (HandleException ex) {
+                    final var identifier = validationResult.status == NODE_DUE_DILIGENCE_FAILURE
+                            ? "node " + creator.nodeId()
+                            : "account " + payer;
+                    logger.error(
```

Comment:
why do we log it and not throw

Reply:
The `TransactionReceipt` would contain the status `FAIL_INVALID` instead of the more specific error code that is available here. What would we gain if we threw an `Exception` here?